### PR TITLE
Add topic vscode-extension

### PR DESCRIPTION
**Pull request by AdamRaichu**
### Please confirm this pull request meets the following requirements:

- [x] I followed the contributing guidelines: <https://github.com/github/explore/blob/main/CONTRIBUTING.md>.
- [x] I have no affiliation with the project I am suggesting (as a maintainer, creator, contractor, or employee).

### Which change are you proposing?

  - [ ] Suggesting edits to an existing topic or collection
  - [x] Curating a new topic or collection
  - [ ] Something that does not neatly fit into the binary options above

---

### Curating a new topic or collection

- [x] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [x] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: <https://github.com/github/explore/tree/main/docs>

This is a topic with 5633 repositories. It should at least have a short blurb explaining what it is.

---

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: topics/vscode-extension/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+display_name: VS Code Extension
+github_url: https://github.com/microsoft/vscode-extension-samples
+related: visual-studio-code
+short_description: An extension for the Visual Studio Code editor
```
Comment (kenyonj) [inline topics/vscode-extension/index.md:5]:
```suggestion
short_description: An extension for the Visual Studio Code editor.

```

**Review by kenyonj — APPROVED** [review]
💯 🍨 Thank you for your contribution! I'll merge this in as soon as CI is 🍏

**Comment by kenyonj** [comment]
Sorry, just realized that this is still in draft!

**Comment by AdamRaichu** [comment]
I left it in draft because I was working in multiple tabs and wanted to make sure I removed the icon. It should be good now.

**Review by kenyonj — APPROVED** [review]
Looks great. Thanks for your contribution!

**Pull request merged**